### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by ImExiled** [AlexGarrixen/vimox-api#65]
Hello,

I apologize if this is an improper question to ask, but I am unable to create a new user in the Database via the Register page or API locally, as it returns "Internal Server Error". This error is only visible when attempting to create a user via CURL.

If you need any information I am happy to provide.

**Comment by AlexGarrixen** [comment]
It is successfully connecting to the database once you run it ?, you should see something like this in console:
`Success connection db`

**Comment by ImExiled** [comment]
> It is successfully connecting to the database once you run it ?, you should see something like this in console: `Success connection db`

Yes, it connects to the Database successfully. I've now found that attempting to add a new account will always result in HTTP500 (Internal Server Error), and I cannot find any way to fix this issue. Attached is a screenshot of the only logging I could find which contains errors, that being of my browser Developer Console.
![vmware_m2ilz09rSL](https://user-images.githubusercontent.com/4431586/163108223-98b4473e-d487-412d-83ca-38f48b657d3c.png)

**Comment by ImExiled** [comment]
Update: Attaching screenshot of Node consoles for reference.
![image](https://user-images.githubusercontent.com/4431586/163110026-84340177-93ea-4eb1-af1d-d1698804e824.png)

**Comment by AlexGarrixen** [comment]
did you set the environment variables ?

**Comment by ImExiled** [comment]
> did you set the environment variables ?

I did, yes. Would it help to see them?

**Comment by AlexGarrixen** [comment]
yes

**Comment by ImExiled** [comment]
> yes

Sure. Two screenshots attached, filenames at top.
![vmware_DeYFJMgWr6](https://user-images.githubusercontent.com/4431586/163271157-7706929a-5874-41ad-9441-7fccec1a35fc.png)

![vmware_0vJqcz29s1](https://user-images.githubusercontent.com/4431586/163271265-bfd2aa06-c432-48cf-99ea-ea40c0f826a8.png)

192.168.209.129 is the internal IP of the machine, and is accessable on my network. Treated as the "publically" accessable domain.

**Comment by AlexGarrixen** [comment]
ok, in development mode you need to set ORIGIN CLIENTS with the url where nextjs serves the app.

 `ORIGIN_CLIENTS=http://localhost:3000`

**Comment by ImExiled** [comment]
I've set that, new error has arisen.
![firefox_hL08nkOiWV](https://user-images.githubusercontent.com/4431586/163275460-5fef4e58-2a92-4aac-a961-e4d1d885715b.png)
 
EDIT: I'm able to sign in now with accounts that didn't previously work!

**Comment by AlexGarrixen** [comment]
send me a screenshot of the network tab

**Comment by ImExiled** [comment]
![firefox_z26dBW8od4](https://user-images.githubusercontent.com/4431586/163280712-b18a5252-9572-4c7c-8881-785048fa49ca.png)
Screenshot attached. Error does still exist in console.

**Comment by AlexGarrixen** [comment]
what is the response from the endpoint POST /auth/signup

**Comment by ImExiled** [comment]
404 if on browser, Internal Server Error via Curl POST request

**Comment by ImExiled** [comment]
Update: Confirmed that even though we receive Internal Server Error, accounts are created. Said accounts *do not* require email verification, for some reason.

**Comment by AlexGarrixen** [comment]
Yes, this happens because the user is not required to check the mail.

It seems that the error could be the mail with which sendgrid sends mail.

Update this part and put your sengrid email
![Annotation 2022-04-13 195933](https://user-images.githubusercontent.com/60531551/163293626-555616b9-0410-4fc1-95f7-677b2b8f16a8.png)

source file: `/src/controllers/auth/signup.ts`

**Comment by ImExiled** [comment]
You were correct, no more errors appear in the Developer Console. I appreciate your help.

The last request I have is to know how to add Genre, Series, and Episodes via API or otherwise. I have direct DB access via Mongo CLI if that works.

**Comment by AlexGarrixen** [comment]
For that I had to make a frontend admin, you could do the same 👍
